WIP: Removing dependency on libvips, updating husky removal, update joplin

sahal opened the pull request:
`libvips` doesn't actually seem necessary anymore.  In fact, it looks like having it installed forces `sharp` to be built.  And for some reason that always fails.  

I also don't see Joplin crashing on my machine (#92) when I remove it from `.SRCINFO` or dependency declarations in `PKGBUILD`.

As I said in #131:

> I'm not sure what the Arch specific way to create a package that requires a specific dependency after its built but fails to build if it that dependency is already installed during the build. This is really a problem when you already have the package installed. Is the Arch standard to remove the dependency before building? Create a chroot to run the package build?

> OTOH, Joplin doesn't seem to crash when I take those out. I'm neither an expert on nodejs apps or Arch packaging. So YMMV.

I also don't use the Joplin CLI.  Something is wrong there, as I can't run it after installing the resultant package.  This is a WIP at best.

sahal committed:
Removing dependency on libvips, updating husky removal

sahal commented on joplin/PKGBUILD:
```diff
@@ -49,11 +48,15 @@ _get_yarn_bin() {
 
     msg2 "Disabling husky (git hooks)"
-    sed -i '/"husky": ".*"/d' "${srcdir}/joplin-${pkgver}/package.json"
+    # This is ugly and unreadable, but it gets rid of "husky" everwhere without breaking package.json
```
Since husky is the last listed dependency in `package.json` now, it breaks when the sed line is run.

See: https://github.com/laurent22/joplin/blob/dev/package.json#L111

sahal committed:
Update .SRCINFO

Updating version

AriaMoradi commented:
Is there a reason to remove husky at all? It only has a pre-commit hook?

alfredopalhares commented:
Thank you for this pull request, specially removing husky has the dependency. 


I will try an look on this tonight.

AriaMoradi commented:
I tried building 2.14.20 with contents of this PR and it failed on building default plugins job, the issue looks similar to https://github.com/laurent22/joplin/issues/9710 It could be fixed with a manual patch though.

sahal committed:
Upgrade to v2.14.20

sahal commented:
I've updated this PR with the latest version: `v2.14.20`.


I built this multiple times using `makechroot` to make sure my system didn't affect the build.

```
export CHROOT=/path/to/temporary/chroot
mkarchroot $CHROOT/root base-devel
arch-nspawn $CHROOT/root pacman -Syu electron gtk3 libexif libgsf libjpeg-turbo libwebp libxss nodejs nss orc rsync git npm yarn python rsync jq yq electron libgsf node-gyp
makechrootpkg -c -r $CHROOT
```

I also rebuilt this without the `-c` (i.e. `clean`) flag to ensure that it worked:
```
makechrootpkg -r $CHROOT
```

Biggest caveat to this PR: I didn't test the resultant package this time. (i.e. for 2.14.20)

I noticed that Joplin complains if I roll back to an older version.  I am too lazy to figure out how to test the package without potentially breaking my own Joplin database.

@AriaMoradi 
Please let me know if the error you were seeing still happens. I didn't see it when using `mkchrootpkg`.  When do you see the error?

I also added a more detailed comment around why Husky is removed.  It boils down to the fact that it tries to add joplin's commit hooks to this repo -- which we don't really want to do.  It probably won't affect any AUR package helper users, but it will annoy the maintainers of this repo.

AriaMoradi commented:
@sahal 
Here's the build log, I used your latest patch, same error as last time, also I'm building with `makepkg -si` 
[build.log](https://github.com/alfredopalhares/arch-pkgbuilds/files/14731783/build.log)

sahal commented:
> @sahal Here's the build log, I used your latest patch, same error as last time, also I'm building with `makepkg -si` [build.log](https://github.com/alfredopalhares/arch-pkgbuilds/files/14731783/build.log)
> 
> It seems like their build process expects a plugin repository to be cloned, fetched, checked out to a specific commit, maybe its using git submodules? I don't find anything special about that in their documentation.

The error is in their build script.  I am no JS dev, but the script should chdir to a git repo holding directory.  The error from git though claims that there is no git repo in that directory. Also, the merged code is in the v2.15.20 branch. 🤔 

* https://github.com/laurent22/joplin/blob/v2.14.20/packages/default-plugins/buildDefaultPlugins.ts#L38-L60

I'm not sure why this is happening for you.  I can't reproduce it.  I pulled the 2.12.15 `PKGFILE` and built the pakage using `makepkg -s`.  Then without running with the `--clean` flag, rebuilt with my branch.  The Joplin package was built successfully. 

I know this is annoying but have you tried, `makepkg --clean`? Or even more annoying: tried clearing yarn cache?

alfredopalhares committed:
Merge branch 'master' into main

sahal commented:
closed, fixed by maintainer elsewhere.
